uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/tables/ColumnMeta.java

```diff
@@ -55,6 +58,12 @@ public ColumnMeta(Column<T, ?> column, String caption,boolean visible, boolean e
         this.extraColumn = extraColumn;
     }
 
+    private void intiDataStoreName(String caption){
+        if(column.getDataStoreName()==null){
```

Comment:
I don't think the ColumnMeta class should be responsible for this. Is actually the ColumnPicker that needs to understand whether to use the caption or dataStoreName. I think a better approach would be to check if  getDataStoreName() is null in the ColumnPicker.